src/app/zap-templates/common/ClustersHelper.js

```diff
@@ -461,78 +554,124 @@ function asPromise(promise)
   return templateUtil.ensureZclPackageId(this).then(fn).catch(err => console.log(err));
 }
 
+//
+// Helpers: Get all
+//
 Clusters.getClusters = function()
 {
     return this.ready.then(() => this._clusters);
 }
 
-Clusters.getCommands = function(name, side)
+Clusters.getCommands = function()
 {
-    const filter = command => command.clusterName.toLowerCase() == name.toLowerCase() && command.clusterSide == side && command.isResponse == false;
+    const filter = command => command.isResponse == false;
     return this.ready.then(() => this._commands.filter(filter));
 }
 
-Clusters.getResponses = function(name, side)
+Clusters.getResponses = function()
 {
-    const filter = command => command.clusterName.toLowerCase() == name.toLowerCase() && command.clusterSide == side && command.isResponse == true;
+    const filter = command => command.isResponse == true;
     return this.ready.then(() => this._commands.filter(filter));
 }
 
-Clusters.getAttributes = function(name, side)
+Clusters.getAttributes = function()
+{
+    return this.ready.then(() => this._attributes);
+}
+
+//
+// Helpers: Get by Cluster Name
+//
+Clusters.getCommandsByClusterName = function(name)
+{
+    const filter = command => command.clusterName.toLowerCase() == name.toLowerCase();
+    return this.getCommands().then(items => items.filter(filter));
+}
+
+Clusters.getResponsesByClusterName = function(name)
+{
+    const filter = command => command.clusterName.toLowerCase() == name.toLowerCase();
```

Comment:
this seems somewhat duplicated with the above and also a bit overlab ith the attribute filter.

Does ZAP require this style of individual namings? I generally would try to avoid copy & paste code (e.g. here the filter applies to responses, yet the filter uses `command` in naming), but unsure what can be suggested here.

Reply:
I have removed the code duplication. `ZAP` by itself does not require anything special.